src/engraving/types/types.h

```diff
@@ -609,6 +610,53 @@ enum class ArpeggioType : unsigned char {
     NORMAL, UP, DOWN, BRACKET, UP_STRAIGHT, DOWN_STRAIGHT
 };
 
+enum class IntervalStep {
+    UNISON,
+    SECOND,
+    THIRD,
+    FOURTH,
+    FIFTH,
+    SIXTH,
+    SEVENTH,
+    OCTAVE
+};
+
+enum class IntervalType {
+    AUTO,
+    AUGMENTED,
+    MAJOR,
+    PERFECT,
+    MINOR,
+    DIMINISHED
+};
+
+struct OrnamentInterval
+{
+    IntervalStep m_step = IntervalStep::SECOND;
+    IntervalType m_type = IntervalType::AUTO;
+
+    OrnamentInterval() = default;
+    OrnamentInterval(IntervalStep s, IntervalType t)
+        : m_step(s), m_type(t) {}
+
+    inline bool operator ==(const OrnamentInterval& interval) const { return m_step == interval.m_step && m_type == interval.m_type; }
+    inline bool operator !=(const OrnamentInterval& interval) const { return !operator ==(interval); }
+
+    bool isPerfect() const
```

Comment:
Perhaps you should add a static function and reuse it here:
https://github.com/musescore/MuseScore/pull/17515/files#r1196125739